### DISCUSSION TITLE
Download the visual model in the front eNotices 2 tool

### DISCUSSION THREAD

**Discussion by AlexAS81** [OP-TED/eForms-SDK#482] [Q&A] [answered]
In addition to being able to download the physical model (from the My notices section) is there a way to download the visual Model in the front eNotices 2 tool?

![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/0b275591-4c83-47fa-9f12-acdf3e0c09d7)

It would help us a lot, since the eForms Notice Editor Demo application does not generate the visual models very well, nor the physical models.

Thanks in advance

**Comment by rouschr** [comment, marked as answer]
eNotices2 is not using exactly the models in the way the Notice Editor Demo does (for historical reasons), but close or getting closer to it.
eNotices2 was not written for demo or teaching purposes and was started before the SDK was a thing (a bit of a chicken and egg situation), but you may be able to intercept their "visual model" by using the web console of your browser when you save or load a notice, that would be my best guess. Note that eNotices2 may be doing various transformations on the server side, that you may never see by just looking at the front end, so even if you get the visual model it could be insufficient for a full understanding of the involved steps.

**Reply by AlexAS81** [reply]
Could you tell us exactly how to capture the visual model with the web console?

Thanks

**Comment by rengit62** [comment]
Thanks you both, very interesting.
Just a more question about it . 
Both eNotice2 and Notice Editor Demo seem to share a common behavior : 
if a subsection is removed and add again ,its counter is incremented. 
For example :  there are 2 Service Provider , 1 and 2, if I delete 2 and add again a Service Provider I will have Service Provider 1 and 3.  
In eNotice2 a counter is added to id , in Notice Editor Demo there is an attribute data-counter.
There is a sort of 'memory' , is this behavior 'wanted' for any reason ? I mean is it necessary to do so ?
Thanks and Regards
Renato
![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/7087cee1-9fc0-482a-9ae7-2d3a5be62d24)
![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/ebb9d211-4711-4b90-b432-21ab2c64f99d)

**Reply by rouschr** [reply]
Yes, if you "forget" the previous ids, you could accidentally recreate previous ids by a sequence of delete, create, delete, create, .... Always incrementing avoids this. We also know there will not be millions of ids for the same kind of object, so we will not run out of them in practice.
